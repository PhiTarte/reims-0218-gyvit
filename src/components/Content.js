import React from "react"

import { Container, Row, Col } from "reactstrap"

import DashboardCard from "./DashboardCard"

const Content = ({ elements, name }) => (
  <Container>
    <Row className="App">
      <Col sm="12" md={{ size: 4, offset: 4 }}>
        <h1>Welcome to Gyvit, {name}.</h1>
      </Col>
      <Col sm="12" md={{ size: 4, offset: 4 }}>
        <h2>What would you like to do ?</h2>
      </Col>
    </Row>
    <Row>
      {elements.map(element => {
        return (
          <Col xs="6" sm="4">
            <DashboardCard key={element.id} contenu={element} />
          </Col>
        )
      })}
    </Row>
  </Container>
)
export default Content
