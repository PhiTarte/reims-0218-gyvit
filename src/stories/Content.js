import React from "react"
import { storiesOf } from "@storybook/react"

import "bootstrap/dist/css/bootstrap.min.css"

import Content from "../components/Content"

const actionHome = [
  {
    id: 0,
    image: "",
    titre: "Documents requests",
    bio:
      "Take the first step to launching your store. Add physical items, digital downloads you can dream up",
    buttonText: "Create first request",
    link: "lien0"
  },
  {
    id: 1,
    image: "",
    titre: "Share documents",
    bio:
      "Take the first step to launching your store. Add physical items, digital downloads you can dream up",
    buttonText: "Share documents",
    link: "lien1"
  },
  {
    id: 2,
    image: "",
    titre: "Store files",
    bio:
      "Take the first step to launching your store. Add physical items, digital downloads you can dream up",
    buttonText: "Upload documents",
    link: "lien2"
  }
]

const name = "Kevin"

storiesOf("Content", module).add("with array of actions", () => (
  <Content elements={actionHome} name={name} />
))
