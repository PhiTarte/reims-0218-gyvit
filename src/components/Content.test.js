import React from "react"
import ReactDOM from "react-dom"
import Content from "./Content"

const elements = [
  {
    id: 0,
    image: "",
    titre: "Documents requests",
    bio:
      "Take the first step to launching your store. Add physical items, digital downloads you can dream up",
    buttonText: "Create first request",
    link: "lien0"
  },
  {
    id: 1,
    image: "",
    titre: "Share documents",
    bio:
      "Take the first step to launching your store. Add physical items, digital downloads you can dream up",
    buttonText: "Share documents",
    link: "lien1"
  },
  {
    id: 2,
    image: "",
    titre: "Store files",
    bio:
      "Take the first step to launching your store. Add physical items, digital downloads you can dream up",
    buttonText: "Upload documents",
    link: "lien2"
  }
]

const name = "Kevin"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<Content elements={elements} name={name} />, div)
  ReactDOM.unmountComponentAtNode(div)
})
