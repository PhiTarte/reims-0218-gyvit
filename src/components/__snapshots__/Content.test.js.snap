// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders link html 1`] = `
<div
  className="container"
>
  <div
    className="App row"
  >
    <div
      className="col-sm-12 col-md-4 offset-md-3"
    >
      <p
        style={
          Object {
            "font-size": "24px",
            "height": "33px",
            "margin-top": "120px",
            "text-align": "center",
          }
        }
      >
        Welcome to Gyvit, 
        Kevin
        .
      </p>
    </div>
    <div
      className="col-sm-12 col-md-4 offset-md-3"
    >
      <p
        style={
          Object {
            "color": "#000000",
            "font-size": "18px",
            "height": "24px",
            "margin-bottom": "60px",
            "opacity": 0.45,
            "text-align": "center",
          }
        }
      >
        What would you like to do ?
      </p>
    </div>
  </div>
  <div
    className="row"
  >
    <div>
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="card"
            style={
              Object {
                "height": "423px",
                "margin-right": "32px",
                "width": "288px",
              }
            }
          >
            <img
              alt="folder"
              className="card-img"
              src="img/folder.ico"
              style={
                Object {
                  "height": "120px",
                  "margin-left": "72px",
                  "margin-right": "64.8px",
                  "margin-top": "31px",
                  "width": "151.2px",
                }
              }
            />
            <div
              className="card-body"
              style={
                Object {
                  "padding": "0px",
                }
              }
            >
              <h5
                className="card-title"
                style={
                  Object {
                    "font-size": "18px",
                    "height": "24px",
                    "margin-left": "57px",
                    "margin-right": "56px",
                    "margin-top": "15px",
                    "padding-left": "0px",
                    "padding-right": "0px",
                    "text-align": "center",
                    "width": "175px",
                  }
                }
              >
                Documents requests
              </h5>
              <p
                className="card-text"
                style={
                  Object {
                    "font-size": "15px",
                    "height": "96px",
                    "line-height": "24px",
                    "margin-bottom": "50px",
                    "margin-left": "30px",
                    "margin-right": "30px",
                    "margin-top": "13px",
                    "opacity": 0.45,
                    "text-align": "center",
                    "width": "228px",
                  }
                }
              >
                Take the first step to launching your store. Add physical items, digital downloads you can dream up
              </p>
              <button
                className="btn btn-secondary"
                onClick={[Function]}
                src="lien0"
                style={
                  Object {
                    "border-radius": "2px",
                    "font-size": "16px",
                    "height": "44px",
                    "margin-bottom": "30px",
                    "margin-left": "59px ",
                    "margin-right": "59px",
                    "padding": "0px",
                    "width": "170px",
                  }
                }
                type={undefined}
              >
                Create first request
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="card"
            style={
              Object {
                "height": "423px",
                "margin-right": "32px",
                "width": "288px",
              }
            }
          >
            <img
              alt="folder"
              className="card-img"
              src="img/folder.ico"
              style={
                Object {
                  "height": "120px",
                  "margin-left": "72px",
                  "margin-right": "64.8px",
                  "margin-top": "31px",
                  "width": "151.2px",
                }
              }
            />
            <div
              className="card-body"
              style={
                Object {
                  "padding": "0px",
                }
              }
            >
              <h5
                className="card-title"
                style={
                  Object {
                    "font-size": "18px",
                    "height": "24px",
                    "margin-left": "57px",
                    "margin-right": "56px",
                    "margin-top": "15px",
                    "padding-left": "0px",
                    "padding-right": "0px",
                    "text-align": "center",
                    "width": "175px",
                  }
                }
              >
                Share documents
              </h5>
              <p
                className="card-text"
                style={
                  Object {
                    "font-size": "15px",
                    "height": "96px",
                    "line-height": "24px",
                    "margin-bottom": "50px",
                    "margin-left": "30px",
                    "margin-right": "30px",
                    "margin-top": "13px",
                    "opacity": 0.45,
                    "text-align": "center",
                    "width": "228px",
                  }
                }
              >
                Take the first step to launching your store. Add physical items, digital downloads you can dream up
              </p>
              <button
                className="btn btn-secondary"
                onClick={[Function]}
                src="lien1"
                style={
                  Object {
                    "border-radius": "2px",
                    "font-size": "16px",
                    "height": "44px",
                    "margin-bottom": "30px",
                    "margin-left": "59px ",
                    "margin-right": "59px",
                    "padding": "0px",
                    "width": "170px",
                  }
                }
                type={undefined}
              >
                Share documents
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="card"
            style={
              Object {
                "height": "423px",
                "margin-right": "32px",
                "width": "288px",
              }
            }
          >
            <img
              alt="folder"
              className="card-img"
              src="img/folder.ico"
              style={
                Object {
                  "height": "120px",
                  "margin-left": "72px",
                  "margin-right": "64.8px",
                  "margin-top": "31px",
                  "width": "151.2px",
                }
              }
            />
            <div
              className="card-body"
              style={
                Object {
                  "padding": "0px",
                }
              }
            >
              <h5
                className="card-title"
                style={
                  Object {
                    "font-size": "18px",
                    "height": "24px",
                    "margin-left": "57px",
                    "margin-right": "56px",
                    "margin-top": "15px",
                    "padding-left": "0px",
                    "padding-right": "0px",
                    "text-align": "center",
                    "width": "175px",
                  }
                }
              >
                Store files
              </h5>
              <p
                className="card-text"
                style={
                  Object {
                    "font-size": "15px",
                    "height": "96px",
                    "line-height": "24px",
                    "margin-bottom": "50px",
                    "margin-left": "30px",
                    "margin-right": "30px",
                    "margin-top": "13px",
                    "opacity": 0.45,
                    "text-align": "center",
                    "width": "228px",
                  }
                }
              >
                Take the first step to launching your store. Add physical items, digital downloads you can dream up
              </p>
              <button
                className="btn btn-secondary"
                onClick={[Function]}
                src="lien2"
                style={
                  Object {
                    "border-radius": "2px",
                    "font-size": "16px",
                    "height": "44px",
                    "margin-bottom": "30px",
                    "margin-left": "59px ",
                    "margin-right": "59px",
                    "padding": "0px",
                    "width": "170px",
                  }
                }
                type={undefined}
              >
                Upload documents
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
